perma_web/static/js/helpers/general.helpers.js

```diff
@@ -31,22 +31,11 @@ export function setCookie (cname, cvalue, exdays) {
     document.cookie = cname + "=" + cvalue + "; " + expires;
 }
 
-
-
-export function getCookie (name) {
-  var cookieValue;
-  if (document.cookie) {
-    var cookies = document.cookie.split(';');
-    for (var i = 0; i < cookies.length; i++) {
-      var cookie = $.trim(cookies[i]);
-      // Does this cookie string begin with the name we want?
-      if (cookie.substring(0, name.length + 1) == (name + '=')) {
-        cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
-        break;
-      }
-    }
-  }
-  return cookieValue;
+export function getCookie(name) {
+  let matches = document.cookie.match(new RegExp(
+      "(?:^|; )" + name.replace(/([\.$?*|{}\(\)\[\]\\\/\+^])/g, '\\$1') + "=([^;]*)"
+  ));
+  return matches ? decodeURIComponent(matches[1]) : undefined;
```

Comment:
I find this easier to reason about, while accomplishing the same thing

Reply:
Wow! This is an impressive regular expression! Did you write it yourself?

I confess that reading that regular expression to figure out what it does is beyond me. If I take it on faith that it returns the first match for a cookie of a given cookie name from the cookie jar, then I agree that `matches ? decodeURIComponent(matches[1]) : undefined;` is very easy to reason about! Whereas I can read the original code and see that it iterates through each cookie, looks at that cookie's name, and see if it's name starts with the name we passed to the utility function.

I have seen folks use very complex regular expressions that others' have written for specific purposes (e.g., credit card number validation, email address validation). I could imagine us doing so here, if there were a clear comment linking to the source of the regular expression and documentation of what it does... and, perhaps, why it is more effective or more efficient or handles edge cases not handled by the string matches, or whatever its advantages may be.

If based on pure readability alone, I think I'd have to argue that the original is more readable.

Reply:
I did not write this RegExp (although I wish I did), my prowess is definitely not that advanced! I had no idea before that you can write a function that dynamically generates a new RegExp pattern. The RegExp was taken from the [Appendix: cookie functions](https://javascript.info/cookie#getcookie-name) section of the JavaScript.info website. 

I absolutely agree — if we decided to keep this function, a comment would be helpful. I could definitely include one that specifically points developers to where the function comes from; explains that `document.cookie` returns a list of cookie values in a semicolon-delimited string of key-value pairs; and that this function dynamically generates new RegEx to match on the name of one specific cookie, using the `; name=<value>` format. 

You're also right that RegExp readability is totally subjective. The reason JavaScript.info provides is that using dynamically-generated RegExp is the fastest way to check a cookie's value, but I am totally impartial and okay with reverting to the original code. (I definitely don't think speed is the most crucial factor here)

Would you strongly prefer the original code?

Reply:
Ah! 

If it's the strategy recommended by JavaScript.info, that's good enough for me 😄: while somewhat inscrutable, we know that source is trustworthy!

There is no equivalent of their reformatting of the cookie name (`name.replace(/([\.$?*|{}\(\)\[\]\\\/\+^])/g, '\\$1')`) in our utility (which is just cut and paste from an old version of [Django's docs](https://docs.djangoproject.com/en/3.0/ref/csrf/), so this is "smarter" on at least one axis.

Why don't we go ahead and keep the regex version, but link out to javascript.info and mention that we are using their recommended approach?

Reply:
That sounds super reasonable! Made a fix for this in [revision 35e6028](https://github.com/harvard-lil/perma/pull/3489/commits/35e602851ac647692ed0475ebe843f3fd76e9016)